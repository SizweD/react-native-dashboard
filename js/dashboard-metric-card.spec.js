import { Text } from 'react-native'
// Note: test renderer must be required after react-native.
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import React from 'react'
import DashboardMetricCard from './dashboard-metric-card'

const createTestProps = props => ({
  title: 'The title',
  iconName: 'chart',
  todayValue: (<Text>{'€ 123'}</Text>),
  weekValue: (<Text>{'€ 123'}</Text>),
  monthValue: (<Text>{'€ 123'}</Text>),
  ...props,
})

describe('rendering', () => {
  let props
  let jsonTree
  beforeEach(() => {
    props = createTestProps()
    jsonTree = renderer.create(
      <DashboardMetricCard {...props} />,
    ).toJSON()
  })
  it('should render title', () => {
    expect(jsonTree).toMatchSnapshot()
  })
})
