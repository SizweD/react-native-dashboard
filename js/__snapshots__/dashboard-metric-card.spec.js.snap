// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with trend should render values with calculated trend 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 6,
      "display": "flex",
      "flex": 1,
      "margin": 8,
      "shadowColor": undefined,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 24,
          }
        }
      >
        The title
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderColor": "#e3e3e3",
              "borderRadius": 3,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
              "overflow": "hidden",
            },
            Object {
              "height": 40,
              "marginLeft": 0,
              "marginRight": 0,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "borderRightColor": "#bdc6cf",
                  "borderRightWidth": 1,
                },
                false,
                Object {
                  "borderBottomLeftRadius": 0,
                  "borderTopLeftRadius": 0,
                },
                Object {
                  "backgroundColor": "white",
                },
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    Object {
                      "color": "#43484d",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Today
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "borderRightColor": "#bdc6cf",
                  "borderRightWidth": 1,
                },
                false,
                false,
                false,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    false,
                    false,
                  ],
                ]
              }
            >
              This week
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                false,
                Object {
                  "borderBottomRightRadius": 0,
                  "borderTopRightRadius": 0,
                },
                false,
                false,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    false,
                    false,
                  ],
                ]
              }
            >
              This month
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontSize": 48,
              }
            }
          >
            0.1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#7ED321",
                  "fontSize": 10,
                },
                undefined,
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#7ED321",
                "paddingLeft": 8,
              }
            }
          >
            100%
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`rendering without trend should render only values 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 6,
      "display": "flex",
      "flex": 1,
      "margin": 8,
      "shadowColor": undefined,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 24,
          }
        }
      >
        The title
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f5f5f5",
              "borderColor": "#e3e3e3",
              "borderRadius": 3,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
              "overflow": "hidden",
            },
            Object {
              "height": 40,
              "marginLeft": 0,
              "marginRight": 0,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "borderRightColor": "#bdc6cf",
                  "borderRightWidth": 1,
                },
                false,
                Object {
                  "borderBottomLeftRadius": 0,
                  "borderTopLeftRadius": 0,
                },
                Object {
                  "backgroundColor": "white",
                },
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    Object {
                      "color": "#43484d",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Today
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "borderRightColor": "#bdc6cf",
                  "borderRightWidth": 1,
                },
                false,
                false,
                false,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    false,
                    false,
                  ],
                ]
              }
            >
              This week
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "flex": 1,
                },
                false,
                Object {
                  "borderBottomRightRadius": 0,
                  "borderTopRightRadius": 0,
                },
                false,
                false,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "#5e6977",
                      "fontSize": 16.25,
                      "fontWeight": "500",
                    },
                    undefined,
                    false,
                    false,
                  ],
                ]
              }
            >
              This month
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 16,
          }
        }
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontSize": 48,
              }
            }
          >
            0.1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
